lambda_functions/transform_lambda.py

```diff
@@ -98,8 +140,17 @@ def process_metric(metric, region, s3_client, s3_prefix, es_client, domain_prefi
         return None
 
 
+@lru_cache(maxsize=256)
```

Comment:
[You can't use `lru_cache` on this function because not all of the arguments are hashable](https://docs.python.org/3/library/functools.html#functools.lru_cache):

> Since a dictionary is used to cache results, the positional and keyword arguments to the function must be [hashable](https://docs.python.org/3/glossary.html#term-hashable).

I think it would be better to extract the function for getting the database instance info to a small function and add lru_cache to that